controllers/istio/istio_controller.go

```diff
@@ -460,15 +461,15 @@ func (r *IstioReconciler) determineStatus(ctx context.Context, istio *v1alpha1.I
 }
 
 func (r *IstioReconciler) updateStatus(ctx context.Context, istio *v1alpha1.Istio, reconcileErr error) error {
+	var errs errlist.Builder
 	status, err := r.determineStatus(ctx, istio, reconcileErr)
-	if err != nil {
-		return err
-	}
+	errs.Add(err)
 
-	if reflect.DeepEqual(istio.Status, *status) {
-		return nil
+	if !reflect.DeepEqual(istio.Status, *status) {
+		err := r.Client.Status().Patch(ctx, istio, kube.NewStatusPatch(*status))
+		errs.Add(err)
```

Comment:
nit: maybe `errs.Add(r.Client.....)`

Reply:
Yeah, I tried it, but it didn't look right. I thought it made it harder to see what was actually happening. But now I've tried it again and it seems fine if wrapped in `errs.Add()`.

Done.